### DISCUSSION TITLE
issue with AuthorizationPolicy when applied on gRPC services

### DISCUSSION THREAD

**Discussion by ravikgvs** [istio/istio#50023] [General]
I am using the latest version of Istio software 16.1. I am seeing an issue with authorizationPolicy resource when used with gRPC services. However the same scenario is working fine with HTTP services. Here are more details about this issue with gRPC:

1. Configured a catch all rule for allow-all:

apiVersion: security.istio.io/v1beta1
kind: AuthorizationPolicy
metadata:
  name: allow-all
  namespace: istio-system
spec:
  rules: 
  - {}

At this point all the end points are allowed. 

2. As per the documentation, I have created this rule for denying access to a specific api endpoint(uthere) by setting the paths field in fully qualified domain name in the form of “/package.service/method”. But it is returning RBAC Access Denied for all the endpoints.  I am following the documentation from this link: https://istio.io/latest/docs/reference/config/security/authorization-policy/

kind: AuthorizationPolicy
metadata:
  name: deny-uthere
  namespace: fs-central
spec:
  selector:
    matchLabels:
      app: example
  action: DENY
  rules:
  - to:
    - operation:
        methods: ["POST"]
        paths: ["/exampleProto.exampleSvc/uthere"]

Do you see any issue with this rule? or Could you please suggest if I am missing anything here? 

Please let me know if any additional information is needed.

Thanks
Ravi

**Comment by ericvn** [comment]
We don't use this repository for project support. See https://istio.io/latest/get-involved/. Please feel free to ask your question on our support forum, https://discuss.istio.io/, if it's still an issue. If it's a bug, see https://istio.io/latest/docs/releases/bugs/.

**Comment by aryan16** [comment]
Check example 3 in 1.17 doc - https://preliminary.istio.io/latest/docs/reference/config/security/authorization-policy/
Ref PR - https://github.com/istio/istio/pull/41802
